app/src/main/java/ro/code4/deurgenta/ui/address/ConfigureAddressFragment.kt

```diff
@@ -0,0 +1,198 @@
+            if (grantResults.all { it == PackageManager.PERMISSION_GRANTED } && grantResults.size == permissions.size) {
+                onPermissionsGranted()
+            } else {
+                onPermissionDenied()
+            }
+        }
+    }
+
+    private fun initMapAndLocationServices(savedInstanceState: Bundle?) {
+        mapView = activity?.findViewById(R.id.map_view)!!
+        mapView.onCreate(savedInstanceState)
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireActivity())
+        mapViewUtils = MapViewUtils(mapView, fusedLocationClient, mapViewCallback)
+    }
+
+    private val mapViewCallback = object : MapViewUtils.MapViewCallback {
+        override fun onError(error: String) {
+            Toast.makeText(requireContext(), error, Toast.LENGTH_SHORT).show()
+        }
+
+        override fun onSuggestionError(error: String) {
+            Log.d(TAG, "error loading autosuggestion results:$error")
+        }
+
+        override fun onAddMarker() {
+            updateSaveButtonVisibility(View.VISIBLE)
+        }
+
+        override fun suggest(suggest: String) {
+            Log.d(TAG, "autosuggestion text.$suggest")
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        mapViewUtils.onResume()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapViewUtils.onPause()
+    }
+
+    private fun onQueryTextListener(view: View): SearchView.OnQueryTextListener {
+        return object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String): Boolean {
+                dismissKeyboard(view)
+                searchOnMap(query)
+                return true
+            }
+
+            override fun onQueryTextChange(queryString: String): Boolean {
+                // Hack
+                if (queryString.isNullOrEmpty()) {
+                    updateSaveButtonVisibility(View.GONE)
+                }
+                searchOnMapAutoSuggestion(queryString)
+                return true
+            }
+        }
+    }
+
+    override fun onPermissionsGranted() {
+        Log.d(TAG, "permission granted.")
+        initMapAndLocationServices(null)
+        initMapButtonCallbacks()
+        mapViewUtils.loadLastKnownLocation(false)
+        initSearchView()
+    }
+
+    override fun onPermissionDenied() {
+        Log.e(TAG, "Permissions denied by user, return back to the configure profile.")
+        findNavController().navigate(R.id.configure_account)
+    }
+
+    private fun updateSaveButtonVisibility(flag: Int) {
+        viewBinding.btSaveAddress.visibility = flag
+    }
+
+    private fun searchOnMap(queryString: String) {
+        mapViewUtils.searchOnMap(queryString)
+    }
+
+    private fun searchOnMapAutoSuggestion(queryString: String) {
+        mapViewUtils.searchOnMapAutoSuggestion(queryString)
+    }
+
+    fun dismissKeyboard(view: View) {
+        hideSoftInput(view)
+    }
```

Comment:
what is the utility of these methods?

Reply:
Basically it is about the hiding of the keyboard when you start the fragment for the first time.

Reply:
I meant why do we need these 3 methods since we could just use the method from maputils/utils. the method from this class do not bring extra functionality